src/Compilers/CSharp/Portable/Compilation/CSharpCompilation.cs

```diff
@@ -2315,7 +2315,8 @@ private static IEnumerable<Diagnostic> FilterDiagnosticsByLocation(IEnumerable<D
             if (stage == CompilationStage.Declare || (stage > CompilationStage.Declare && includeEarlierStages))
             {
                 var declarationDiagnostics = GetSourceDeclarationDiagnostics(syntaxTree, filterSpanWithinTree, FilterDiagnosticsByLocation, cancellationToken);
-                Debug.Assert(declarationDiagnostics.All(d => d.HasIntersectingLocation(syntaxTree, filterSpanWithinTree)));
+                // re-enable the assert once https://github.com/dotnet/roslyn/issues/21185 is fixed
+                // Debug.Assert(declarationDiagnostics.All(d => d.HasIntersectingLocation(syntaxTree, filterSpanWithinTree)));
```

Comment:
I can't remember exact case, but I think it is compiler bug since we are just directly calling SemanticModel.GetDiagnostics(span) in test code, and it asserts. one can get live repro by running IOperationTests.TestClone after re-enabling the assert.